src/storage/table_factory.cpp

```diff
@@ -19,7 +19,6 @@
 #include "storage/storage_manager.h"
 #include "storage/temp_table.h"
 
-#include <mutex>
```

Comment:
I think you can remove `#include "index/index.h"` and `#include "common/logger.h"`, too.